sicp-pict/main.rkt

```diff
@@ -396,10 +414,7 @@
 
 ;;; Number Painter
 ;;;     A number painter is a color painter that draws a gray color from 0 to 255.
-(define (number->painter number-or-color)
-  (define n number-or-color)
-  (unless (and (number? n) (<= 0 n 255))
-    (raise-type-error 'number->painter "number between 0 and 255" n))
+(define (number->painter n)
   (color->painter (new-color n)))
 
```

Comment:
The error produced here will refer to `new-color` instead of `number->painter`.
Thus a beginner won't know where to look for the error.

Reply:
I thought the error would be raised by the contract, so `raise-type-error` didn't work here.

Reply:
Oh! I had forgotten about the contract (it wasn't shown in the diff).